embed: allow embeds like YouTube to link back to its site

mimecuvalo opened the pull request:
Fixes https://github.com/tldraw/tldraw/issues/3593

(some drive-by spelling/grammar cleanup)

### Change Type

<!-- ❗ Please select a 'Scope' label ❗️ -->

- [x] `sdk` — Changes the tldraw SDK
- [ ] `dotcom` — Changes the tldraw.com web app
- [ ] `docs` — Changes to the documentation, examples, or templates.
- [ ] `vs code` — Changes to the vscode plugin
- [ ] `internal` — Does not affect user-facing stuff

<!-- ❗ Please select a 'Type' label ❗️ -->

- [x] `bugfix` — Bug fix
- [ ] `feature` — New feature
- [ ] `improvement` — Improving existing features
- [ ] `chore` — Updating dependencies, other boring stuff
- [ ] `galaxy brain` — Architectural changes
- [ ] `tests` — Changes to any test code
- [ ] `tools` — Changes to infrastructure, CI, internal scripts, debugging tools, etc.
- [ ] `dunno` — I don't know


### Release Notes

- Embeds: fix being able to click on links that go back to the embed's site (e.g. YouTube)

steveruizok reviewed (commented):
I think individual embeds can override these, right? Should we make that opt in on in the YT links config rather than changing the default?

mimecuvalo commented:
> I think individual embeds can override these, right? Should we make that opt in on in the YT links config rather than changing the default?

sounds good. I think i went for the general one because it affects all google properties.

i changed maps/cal/slides/youtube now to override. 

also, drive-by fix: i noticed that val.town was busted. i fixed that up too.
